File: octopoints_flutter/lib/service/rule_service.dart
import 'package:octopoints_flutter/service/service.dart';

abstract class RuleService {
  Future<Rule> createRule(Rule rule);
  Future<int> updateRule(Rule rule);
  Future<int> deleteRule(Rule rule);
  Future<Rule?> getRuleByMatchId(int id);
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/RuleService.kt
package dev.tpavani.octopoints.service

import dev.tpavani.octopoints.service.model.Rule

interface RuleService {
    suspend fun createRule(rule: Rule): Rule
    suspend fun deleteRule(rule: Rule)
    suspend fun updateRule(rule: Rule)
    suspend fun getRuleByMatchId(id: Int): Rule?
}